File: PreUtil/src/output/java/easy/ContainsDuplicateIi.java
/**https://leetcode.com/problems/contains-duplicate-ii/ */
//Given an integer array nums and an integer k, return true if there are two distinct indices i and j in the array such that nums[i] == nums[j] and abs(i - j) <= k.


class ContainsDuplicateIi {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            if(map.containsKey(nums[i])){
                if(i - map.get(nums[i]) <= k) return true;
            }
            map.put(nums[i], i);
        }
        return false;
        
        
    }
}

    

File: Tool/PreUtil/src/output/c/easy/ContainsDuplicateIi.c
/**https://leetcode.com/problems/contains-duplicate-ii/ */
//Given an integer array nums and an integer k, return true if there are two distinct indices i and j in the array such that nums[i] == nums[j] and abs(i - j) <= k.

bool containsNearbyDuplicate(int* nums, int numsSize, int k){
    int map[100000] = {0};
    for(int i = 0; i < numsSize; i++){
        if(map[nums[i]] != 0 && i + 1 - map[nums[i]] <= k){
            return true;
        }
        map[nums[i]] = i + 1;
    }
    return false;

}
    